src/profile-logic/profile-data.js

```diff
@@ -904,6 +904,29 @@ export function filterThreadSamplesToRange(
   thread: Thread,
   rangeStart: number,
   rangeEnd: number
+): Thread {
+  const { samples } = thread;
+  const [sBegin, sEnd] = _getSampleIndexRangeForSelection(
+    samples,
+    rangeStart,
+    rangeEnd
+  );
+  const newSamples = {
+    ...samples,
+    stack: samples.stack.map((stackIndex, sampleIndex) =>
+      sampleIndex >= sBegin && sampleIndex < sEnd ? stackIndex : null
+    ),
+  };
+  return {
+    ...thread,
+    samples: newSamples,
+  };
+}
+
+export function trimThreadSamplesToRange(
```

Comment:
I kept the old implementation, simply renaming it, for the sanitization (as you'll see below)